pkg/flare/archive.go

```diff
@@ -339,6 +339,9 @@ func zipExpVar(tempDir, hostname string) error {
 	if config.Datadog.IsSet("apm_config.receiver_port") {
 		apmPort = config.Datadog.GetString("apm_config.receiver_port")
 	}
+	if v := os.Getenv("DD_APM_RECEIVER_PORT"); v != "" {
```

Comment:
This needs to go before the previous `if` because the file settings will override the environment variables when set. This is how all of the config works. Also, normally `apm_config.receiver_port` should have already gotten the value from the environment, but because of a bug it doesn't, so we have to leave a comment here to remember to fix this:
```go
// TODO(gbbr): Remove this once we use BindEnv for trace-agent
if v := os.Getenv("DD_APM_RECEIVER_PORT"); v != "" {
   // ...
```
I will know to address it.

Reply:
Which bug ?

Reply:
@hush-hush it's not a "bug" per se. It's a longer story. Short version is that environment variables used by APM are bound in the trace-agent, and not in the main config package using `BindEnv`. This might ring a bell from another PR. I already have a task to address this for the next release. As part of that work I will also remove this line (that's why I asked for the `TODO` note).

Reply:
Thanks for the info.
So, don't we have an issue here when each agent is running in a different container (ie: as we do in our official heml chart). If the core-agent generate the flare it might now have the same env vars than the trace-agent.
I don't think it's an issue but we should loop with the container team to, maybe, update the helm chart/documentation to specify this.

WDYT ?

Reply:
Hi, 
Yes all the environment are not shared between very containers in the helm chart.
I feel like the "communication" cross agent can help here: the core-agent ask other agent to report their flares informations.

Reply:
We already have the ability to generate the flare locally (like when the agent is not running). We should document that to get the trace flare, in our helm chart, we need to use this option.

Reply:
@clamoriniere @hush-hush could you approve this Pr?

Reply:
@clamoriniere @hush-hush Could you approve this PR?